### PR TITLE
Improvements to local development experience

### PR DESCRIPTION
**Pull request by timwright12**
Added new script to allow local viewing of vets-website generated assets through a symlink inside the content-build repo.

Testing Instructions:
1) remove `./build`
2) run `yarn build` or `yarn watch` (if you run build you'll need to start a server in ./build/localhost/)
3) Visit the site and notice there are no styles
5) Refresh the site, it should be styled now, as the symlink is created

Running `yarn watch:content` should spin up a site at localhost:3002; if you roll your own, it just can't be on 3001, because vets-website uses it.

Added a flag to not process the symlink just in case it ever gets in the way: `yarn build --noSymlink`

### CODE REVIEW

**Comment by timwright12** [comment]
@billfienberg this is ready for review/testing

**Comment by timwright12** [comment]
@billfienberg pulling this back to WIP since we need the process-entry-names plugin working to view the CSS/JS and if fails the build

File: src/site/stages/build/index.js [hunk 1]
```diff
@@ -24,16 +22,17 @@ const addSubheadingsIds = require('./plugins/add-id-to-subheadings');
 const checkBrokenLinks = require('./plugins/check-broken-links');
 const checkCollections = require('./plugins/check-collections');
 const checkForCMSUrls = require('./plugins/check-cms-urls');
-// const downloadAssets = require('./plugins/download-assets');
+const downloadAssets = require('./plugins/download-assets');
 // const readAssetsFromDisk = require('./plugins/read-assets-from-disk');
-// const processEntryNames = require('./plugins/process-entry-names');
+const processEntryNames = require('./plugins/process-entry-names');
```
Comment (timwright12) [inline src/site/stages/build/index.js:27]: we still need this to generate templates for the apps

File: src/site/stages/build/index.js [hunk 2]
```diff
@@ -24,16 +22,17 @@ const addSubheadingsIds = require('./plugins/add-id-to-subheadings');
 const checkBrokenLinks = require('./plugins/check-broken-links');
 const checkCollections = require('./plugins/check-collections');
 const checkForCMSUrls = require('./plugins/check-cms-urls');
-// const downloadAssets = require('./plugins/download-assets');
+const downloadAssets = require('./plugins/download-assets');
```
Comment (timwright12) [inline src/site/stages/build/index.js:25]: Downloading assets in a non-localhost environment (localhost uses symlink)

File: src/site/stages/build/index.js [hunk 3]
```diff
@@ -111,10 +61,14 @@ function build(BUILD_OPTIONS) {
     enabledFeatureFlags: BUILD_OPTIONS.cmsFeatureFlags,
   });
 
-  smith.use(
-    preserveWebpackOutput(BUILD_OPTIONS.destination, BUILD_OPTIONS.buildtype),
-    'Preserving Webpack build output',
-  );
+  // If you're on localhost, you probably want to see CSS/JS reflected in the build,
+  // so, this will set up a symlink into vets-website for you.
+  if (BUILD_OPTIONS.buildtype === 'localhost' && !BUILD_OPTIONS.nosymlink) {
```
Comment (timwright12) [inline src/site/stages/build/index.js:66]: There's also an option for `--nosymlink` to download dev assets instead

File: src/site/stages/build/plugins/download-assets.js [hunk 4]
```diff
@@ -2,47 +2,45 @@
 /* eslint-disable no-console */
 
 require('isomorphic-fetch');
-const https = require('https');
-const fs = require('fs-extra');
-const path = require('path');
-const decompress = require('decompress');
+// const https = require('https');
```
Comment (timwright12) [inline src/site/stages/build/plugins/download-assets.js:5]: Just commenting out all this stuff for now, will circle back later after we can confirm a deploy to dev is working

**Comment by timwright12** [comment]
@billfienberg passing Jenkins and ready for review/testing now

**Comment by billfienberg** [comment]
Nice

```
/**
 * ATTN: Cannot create symlink with vets-website.
 * Path: /Users/bill/work/vets-website/build/localhost/generated does not exist.
 * Please run "yarn build" in vets-wesbite to create this directory.
 */
```
image

**Review by billfienberg — APPROVED** [review]
image

**Pull request merged**